### DISCUSSION TITLE
Can't override value from style

### DISCUSSION THREAD

**Discussion by mohdimas** [AvaloniaUI/Avalonia#13639] [Q&A]
hi there,
it seems that since updating to avalonia 11.0.5, I can no longer override a property of a control inline. It'll always pick the value from the style. So for example:

```
<Style Selector="Path.DownArrowBoldPath">
        <Setter Property="Width"
                Value="12" />
        <Setter Property="Height"
                Value="12" />
        <Setter Property="Data"
                Value="..." />
    </Style>

...

<Path Classes="DownArrowBoldPath"
           Width="20"
           Height="20" />

```

Then the path will have width and height of 12. is there any known issue? many thanks

**Comment by timunie** [comment]
please share path data with us. Otherwise we cannot investigate it properly.

**Reply by mohdimas** [reply]
It shouldn't be related with the path, but here you go
```
<Setter Property="Data"
                Value="M0 0 M12 12 M10.7071 4.70711C11.0976 4.31658 11.0976 3.68342 10.7071 3.29289C10.3166 2.90237 9.68342 2.90237 9.29289 3.29289L6 6.58579L2.70711 3.29289C2.31658 2.90237 1.68342 2.90237 1.29289 3.29289C0.902369 3.68342 0.902369 4.31658 1.29289 4.70711L5.29289 8.70711C5.34171 8.75592 5.39432 8.79864 5.44977 8.83525C5.83792 9.09153 6.3654 9.04882 6.70711 8.70711C6.70691 8.7073 6.7073 8.70691 6.70711 8.70711L10.7071 4.70711Z" />
```

**Reply by timunie** [reply]
it can be related to the path as if the path is only 12, the result will not grow unless you set Stretch-mode to request it

**Reply by timunie** [reply]
The path is 12x12: 
![image](https://github.com/AvaloniaUI/Avalonia/assets/47110241/0175c181-8cea-4ded-88f0-b22b16197cf7)

**Reply by mohdimas** [reply]
yes but the path control itself should have the correct bound when checked via the dev tools.

**Reply by mohdimas** [reply]
Here's from the dev tools

<img width="1392" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/1266072/74dd22c8-6ff2-4256-ac0f-693be1edb97c">

and here's the path control
```
<Path Width="20"
    Height="20"
    Classes="DownArrowBoldPath" />
```

**Reply by timunie** [reply]
Please add `<Setter Property="Stretch" Value="Uniform" />` to your style. This works perfectly for me. 

> [!IMPORTANT]  
> By default stretch is None, which means if the path is 12 px it will not auto grow. Unifrom will size it acc. to the size you sel.

**Reply by mohdimas** [reply]
Thanks, but it doesn't work for my case. It's still using 12 for both width and height not 20

<img width="361" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/1266072/d28adb29-14ff-4474-8384-14035129db07">

<img width="546" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/1266072/f1cbe55c-4b1d-43eb-b68b-c6575770a499">

<img width="539" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/1266072/68044349-532d-40df-97ce-e0b4664f8c95">

<img width="408" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/1266072/606eed16-5db3-436f-b624-f1c226d02a07">


This is just an example which I believe isn't related to the path. another example: 

```
<Style Selector="ContentControl.DefaultPopupPanel">
        <Setter Property="Background"
                Value="{DynamicResource MenuBackgroundBrush}" />
        <Setter Property="BorderBrush"
                Value="{DynamicResource MenuBorderBrush}" />
        <Setter Property="BorderThickness"
                Value="{StaticResource ControlBorderThickness}" />
        <Setter Property="Padding"
                Value="16,8" />
</Style>
...
<ContentControl Classes="DefaultPopupPanel"
         Padding="0">
</ContentControl>
```

The padding used in the `ContentControl` is `16,8` and not `0`. There are few other examples. We have a workaround by adding local style, and then it'll override it correctly. But I just want to point out that there must be some issues as it worked in the Avalonia 0.10.x.

**Reply by timunie** [reply]
yeah we had breaking changes between 11.0 and 0.10. But still, In the test case it just worked. So please attach a minimal sample if you think it is a bug. Without being able to reproduce, we cannot finally judge.

**Reply by mohdimas** [reply]
ok here's the [sample](https://github.com/mohdimas/AvaloniaTest). It looks like the style can't be overridden when it's inside a `ContentControl`. Both path on the left and right have the same properties overriding the original style. But as you can see the one on the left still uses the width and height from the original style.

<img width="1462" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/1266072/37f4bbb4-1254-46d5-810e-25184c06086a">

**Reply by timunie** [reply]
that's interesting 👀

**Reply by timunie** [reply]
made it an issue for now because I think there is smth strange going on.

**Comment by maxkatz6** [comment]
https://docs.avaloniaui.net/docs/next/guides/styles-and-resources/troubleshooting

**Reply by timunie** [reply]
> Locally Set Properties Have Priority

that is what I do not understand in this case to be honest. I thought it was handled as a local set value? or does it count as "TemplatedParent" because it is defined in a ControlTemplate 🤔 ?

https://github.com/mohdimas/AvaloniaTest/blob/91b4f932cf3d833d33c62ea01a2549efcd968eb3/Resources.xaml#L3-L10

```xml
<ControlTemplate x:Key="SplashScreenTemplate">
    <Panel x:Name="SplashScreenPanel">
        <Path Classes="MailbirdPath"
              Width="150"                    ◄ should not this be 150? And have higher priority than the Style-Setter?
              Height="150"                   ◄ should not this be 150? And have higher priority than the Style-Setter?
              VerticalAlignment="Center"
              HorizontalAlignment="Center"
              Fill="Red" />
```

**Reply by maxkatz6** [reply]
Ah, so it's the opposite. I need to check the sample on the week

**Reply by MrJul** [reply]
> that is what I do not understand in this case to be honest. I thought it was handled as a local set value? or does it count as "TemplatedParent" because it is defined in a ControlTemplate

Yes, that value has a `Template` priority. If controls in templates weren't special cased with lower priorities, one wouldn't be able to override all their properties using a `Style` (a thing people are doing all the time with built-in themes.)

**Reply by mohdimas** [reply]
> one wouldn't be able to override all their properties using a Style

This is the opposite. We can't override style with the locally set values in a control. But even that is not consistent. In the attached repository, `Fill` can be overridden but not `Width` and `Height` when the control is within a `ControlTemplate`.
